Add k6 integration

ppcano opened the pull request:
### What does this PR do?

It adds the integration of the [k6 load testing tool](https://k6.io/open-source) to Datadog. 

### Motivation

The k6 community has requested multiple times to send k6 metrics to Datadog to be able to analyze and visualize k6 performance testing metrics and correlate these metrics with other monitored metrics in Datadog. 

The instructions of this integration are available at the [k6 documentation](https://k6.io/docs/getting-started/results-output/datadog).

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

LoadImpact, the company behind k6, is a Datadog technology partner. The Datadog Technology Partner Development guide states:

> All integrations are expected to contain an out-of-the-box dashboard. This dashboard will be loaded by default when users install the integration. It serves as an example of the best way to display relevant metrics for your product.

> Once you have created a dashboard in your sandbox account that you would like to include with the integration, include the dashboard name with your pull request (this process is outlined in Step 5)

We have included in this PR a default dashboard but I am not sure what the next step is.

hithwen commented:
Hallo, thanks for your contribution! You need to include the integration and yourself in `.github/CODEWNERS` file.

Meanwhile we'll start the process of creating an in app tile for this integration.

hithwen commented:
/azp run

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+For the detailed instructions, follow the [k6 documentation][2]
+
+### Step 1 - Copy your Datadog API Key
```
```suggestion
### Installation

1. In Datadog, navigate to Integrations > [API][3] to copy your API key.
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+### Step 1 - Copy your Datadog API Key
+
+In Datadog, navigate to [Integrations --> API][3] and copy your API Key.
```
```suggestion
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+In Datadog, navigate to [Integrations --> API][3] and copy your API Key.
+
+### Step 2 - Run the Datadog Agent
```
```suggestion
2. Run the Datadog Agent:
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+### Step 2 - Run the Datadog Agent
+
+To get k6 metrics into Datadog, k6 has to send metrics to the Datadog Agent that will collect, aggregate, and forward the metrics to the Datadog platform.
```
```suggestion
To get k6 metrics into Datadog, k6 sends metrics through the Datadog Agent, which collects, aggregates, and forwards the metrics to the Datadog platform.
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+To get k6 metrics into Datadog, k6 has to send metrics to the Datadog Agent that will collect, aggregate, and forward the metrics to the Datadog platform.
+
+You can run the Datadog Agent service as a Docker container with this command:
```
```suggestion
Run the Datadog Agent service as a Docker container with this command:
```

ruthnaebeck commented on k6/README.md:
````diff
@@ -0,0 +1,86 @@
+    datadog/agent:latest
+```
+
+Note that you have to replace `<YOUR_DATADOG_API_KEY>` with your API key, and if your account is registered with Datadog EU, the value of `DD_SITE` should be `datadoghq.eu`.
````
```suggestion
**Note**: Replace `<YOUR_DATADOG_API_KEY>` with your [API][3] key. If your account is registered with Datadog EU, change the value of `DD_SITE` to `datadoghq.eu`.
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+Note that you have to replace `<YOUR_DATADOG_API_KEY>` with your API key, and if your account is registered with Datadog EU, the value of `DD_SITE` should be `datadoghq.eu`.
+
+### Step 3 - Run the k6 test and output the results to Datadog
```
```suggestion
3. Run the k6 test and output the results to Datadog.
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+### Step 3 - Run the k6 test and output the results to Datadog
+
+Once the Datadog Agent service is running, you can run the k6 test and send the metrics to the agent with:
```
```suggestion
Once the Datadog Agent service is running, run the k6 test and send the metrics to the Agent with:
```

ruthnaebeck commented on k6/README.md:
````diff
@@ -0,0 +1,86 @@
+k6 run --out datadog script.js
+```
+
+### Step 4 - Visualize the k6 metrics in Datadog
````
```suggestion
4. Visualize the k6 metrics in Datadog.
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+### Step 4 - Visualize the k6 metrics in Datadog
+
+While running the test, k6 send metrics periodically to DataDog. By default, these metrics have `k6.` as name prefix. 
```
```suggestion
While running the test, k6 sends metrics periodically to DataDog. By default, these metrics have `k6.` as the name prefix. 
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+While running the test, k6 send metrics periodically to DataDog. By default, these metrics have `k6.` as name prefix. 
+
+You can visualize k6 metrics in realtime at the [Datadog Metrics Explorer][5], creating [monitors][6] or [custom dashboards][7] for your performance testing metrics.
```
```suggestion
You can visualize k6 metrics in realtime with the [metrics explorer][5], [monitors][6], or [custom dashboards][7].
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+### Service Checks
+
+k6 does not include any service checks.
```
```suggestion
The k6 integration does not include any service checks.
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+### Events
+
+k6 does not include any events.
```
```suggestion
The k6 integration does not include any events.
```

ruthnaebeck commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+
+## Troubleshooting
+
+Need help? Read the [k6 Datadog documentation](2) or contact the [k6 support][10].
```
```suggestion
Need help? Read the [k6 Datadog documentation](2) or contact [k6 support][10].
```

ruthnaebeck commented on k6/manifest.json:
```diff
@@ -0,0 +1,33 @@
+    "notification"
+  ],
+  "type": "check",
+  "is_public": false,
```
```suggestion
  "is_public": true,
```

ppcano commented on k6/README.md:
```diff
@@ -0,0 +1,86 @@
+- Create alerts based on performance testing metrics.
+- Analysis and visualize k6 metrics using the k6 Datadog Dashboard or [Metrics Explorer][5].
+
+![k6 Datadog Dashboard][9]
```
Fixed by https://github.com/DataDog/integrations-extras/pull/578/commits/0c4815e773528892763e1f305d594438e40c39b6.

Thanks, @ruthnaebeck for the review.

ppcano commented:
Hola @hithwen,

I have addressed the reported issues:  

- Added info to `CODEOWNERS` at https://github.com/DataDog/integrations-extras/pull/578/commits/fc77e1c5c25a2e7b66ea4bdfa00b7ad40bd3920f
- Addressed @ruthnaebeck's review (https://github.com/DataDog/integrations-extras/pull/578/commits/0c4815e773528892763e1f305d594438e40c39b6 and https://github.com/DataDog/integrations-extras/pull/578/commits/68e47aaaaa6b480694b253bd47c91b67ed90c76f)
- The Dashboard uses the format provided by the [Dashboard API](https://docs.datadoghq.com/api/?lang=bash#update-a-dashboard) at https://github.com/DataDog/integrations-extras/pull/578/commits/a095b16820943dd297d3f8e8f01cd5510df71b27

ruthnaebeck commented on k6/README.md:
````diff
@@ -0,0 +1,86 @@
+```
+
+**Note**: Replace `<YOUR_DATADOG_API_KEY>` with your [API][3] key. If your account is registered with Datadog EU, change the value of `DD_SITE` to `datadoghq.eu`.
+
````
Indent content below numbers 2, 3, and 4 by 4 spaces. This will make it display properly on the Datadog docs site.

ppcano replied:
It should be fixed by https://github.com/DataDog/integrations-extras/pull/578/commits/7f7cda9f78aeb6e3258fdf2ad9d82f2c47d2fe8e. Let me know otherwise, thanks.

hithwen commented:
Seems the dashboard error is still in place:
```
    /home/vsts/work/1/s/k6/assets/dashboards/overview.json is using the new /dash payload format which isn't currently supported. Please use the format from the /screen or /time API endpoints instead.
```

You need to use the [screen](https://docs.datadoghq.com/dashboards/guide/screenboard-api-doc/?tab=python) or [time](https://docs.datadoghq.com/api/?lang=bash#timeboards) APIs instead

ppcano commented:
@hithwen 

The [screen API](https://docs.datadoghq.com/dashboards/guide/screenboard-api-doc/?tab=python) returns an empty array. The reason might be that the k6 Dashboard is a Timeboard.

The [time API](https://docs.datadoghq.com/api/?lang=bash#timeboards) is outdated and I cannot find the API endpoint.  

> This endpoint is outdated. Use the new Dashboard endpoint instead. Documentation for old endpoints is available in the Screenboard API documentation.

The new dashboard API was used to return the current `overview.json` file.

How should I proceed?

hithwen commented:
Hi @ppcano, default dashboards need to be screenboards, so the options are to either redo the dashboard as screenboard or offer the integration without a default dashboard (ie: remove it from the PR). Sorry for the trouble!

ppcano commented:
@hithwen,  it should be done now at https://github.com/DataDog/integrations-extras/pull/578/commits/0326fdb54d481640868ef7a64ad67351c76b6dd3.

hithwen commented on k6/manifest.json:
```diff
@@ -0,0 +1,33 @@
+  "metric_prefix": "k6.",
+  "metric_to_check": "",
+  "creates_events": false,
+  "short_description": "Collect metrics from k6 to visualize and correlate performance testing metrics in DataDog",
```
Short description is 80 characters max :(

ppcano replied:
@hithwen Done at https://github.com/DataDog/integrations-extras/pull/578/commits/e6eba49676c8be4fb1e8e9973940527888c16312
